core/src/util/usUtils.cpp

```diff
@@ -388,46 +388,43 @@ std::string GetLastErrorStr()
 #endif
 }
 
-static MsgHandler handler = 0;
-
-MsgHandler installMsgHandler(MsgHandler h)
-{
-  MsgHandler old = handler;
-  handler = h;
-  return old;
-}
-
-void message_output(MsgType msgType, const char *buf)
+void TerminateForDebug(const std::exception_ptr ex)
```

Comment:
I agree with you, there is no sensible place for terminating the process. I somewhat overlooked the "termination" part and was mostly thinking about the ability to trigger a debug break, if a particular framework property is set. This could be nice to have in certain scenarios. For assertions, I believe we should just use the existing ` assert` macro.

What about moving this discussion into a new issue and move forward with the PR?